flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+    }
+
+    def notify(self, title, message, config):
+        """
+        Publish to an MQTT topic
+        """
+        try:
+            import paho.mqtt.client as mqtt
+            from hashlib import sha256
+            import ssl
+            import time
+
+        except ImportError as e:
+            logger.verbose('Error importing paho.mqtt.client: {}', e)
+            raise DependencyError(
+                plugin_name, 'paho.mqtt.client', 'paho-mqtt python module is required for MQTT notify plugin. ImportError: %s' % e
+            )
+
+        config['title'] = title
+        config['message'] = message
+        config['payload'] = '{} - {}'.format(config.get('title'),config.get('message'))
+
+        class PublishMQTT(mqtt.Client):
+            conn_rc_description_map = { 0: 	'Connection Accepted',
+                                1: 	'Connection Refused, unacceptable protocol version - The Server does not support the level of the MQTT protocol requested by the Client',
+                                2: 	'Connection Refused, identifier rejected - The Client identifier is correct UTF-8 but not allowed by the Server',
+                                3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                                4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                                5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+            MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
+                               'MQTTv311': 	mqtt.MQTTv31 }
+
+            def __init__(self, config):
+
+                try:
+                    self.config = config
+                    self.logger = logger
+
+                    blank2None = lambda s: s or None
+                    self.config.update({'username': blank2None(self.config.get('username'))})
+                    self.config.update({'password': blank2None(self.config.get('password'))})
+                    self.config.get('certificates',{}).update({'broker_ca_cert': blank2None(self.config.get('certificates',{}).get('broker_ca_cert'))})
+                    self.config.get('certificates',{}).update({'client_cert': blank2None(self.config.get('certificates',{}).get('client_cert'))})
+                    self.config.get('certificates',{}).update({'client_key': blank2None(self.config.get('certificates',{}).get('client_key'))})
+                    self.config.get('certificates',{}).update({'tls_version': blank2None(self.config.get('certificates',{}).get('tls_version'))})
+                    logger.trace('MQTT notify config={}'.format(str(self.config)))
+
+                    mqtt.Client.__init__(self, protocol=PublishMQTT.MQTT_proto_map.get(self.config.get('broker_protocol',mqtt.MQTTv311)), transport=self.config.get('broker_transport', 'tcp') )
```

Comment:
please user `super()` for parent calls, and `self` to call instance/class methods from the class instance

Reply:
no longer an issue after refactoring code - commit 964dba7624322c86701376d4875a23025a9b8bf8